### PR TITLE
I47 connect login modal to backend

### LINKED ISSUE #47: Connect Login modal to backend
## Basic Information
Create an Auth pinia store
- the store should keep track of the username, and the JWT token. We can add more data later
- You can call the JWT endpoint api/token with the username and password, this request will return a token if it's a valid user.
- Look into axios to handle requests

NOTE: If you run into an error running the server docker container, you might need to rebuild containers, volumes and images.

### PR DESCRIPTION
**Pull request by Dhruv8601**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**
Created userstore for user
Calling jwt token endpoint when logging in
If successful then stores the username else returns error
Added cors-headers to backend
Changed user_id to id

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**
The current login form shows email and password so needs to be merged with the new login form in issue #46 and needs to display as username and password in the form. I've been told new login form is completed but not merged to main yet.

I added cors-headers using poetry so people might (not sure) need to rebuild their docker image to get the package.

Resolve #47 

# Related Issue

- Resolve #47

### CODE REVIEW

File: client/src/stores/user.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        await axios
+          .post(`${baseUrl}/auth/token/`, {
+            username: username,
+            password: password
+          })
+          .then((res) => {
+            if (res.status == 200) {
+              this.authUser = username
+            }
+          })
```
Comment (ChantelleKerr) [inline client/src/stores/user.ts:25]: We will also need to store the token so we can use it to access some of our private endpoints

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
Nice job, I'm just not sure where the token is stored.

File: client/src/stores/user.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { defineStore } from 'pinia'
+import axios from 'axios'
+
+const baseUrl = 'http://localhost:8081/api'
+
+export const useUserStore = defineStore('user', {
+  state: () => ({
+    authUser: null as string | null
+  }),
```
Comment (ChantelleKerr) [inline client/src/stores/user.ts:9]: State doesnt persistent when refreshing and opening in a new tab. Pinia should have some options to help with this :)

File: client/src/stores/user.ts (hunk 1 above)
Comment (Dhruv8601) [reply client/src/stores/user.ts]: Right, I wasn't sure if we were supposed to store that. The token has two values, a refresh and an access value. I added a token variable that stores both of these in a JSON format.

File: client/src/stores/user.ts (hunk 2 above)
Comment (Dhruv8601) [reply client/src/stores/user.ts]: Added new package @vueuse/core which gives some localstorage functionality.

**Review by ChantelleKerr — APPROVED** [review]
Great good! Looks good to me :)

Approved. You can merge it in now :)

**Pull request merged**